Update doc to reflect that Solr8 is supported (#1197)
ezsystems/developer-documentation: docs/guide/search/solr.md

## How to set up Solr search engine

!!! note "Installing the bundle"

    If you have previously removed the bundle, add/update composer dependencies:
    ``` bash
    composer require --no-update ezsystems/ezplatform-solr-search-engine:~3.0
    composer update
    ```
    

>     Symfony Flex will enable the bundle for you when installing the package.

Author: @dabrt FYI reverted part of this for 2.5 in 7cabc6801085a1ef16ac2a24e1c274648cf8e100, I forgot to mention the changes like this was meant for 3.x only.

>     Symfony Flex should enable the bundle for you when installing the package.

Reviewer: ```suggestion
    Symfony Flex should enable the bundle for you when installing the package.
```
"Should enable" or "will enable"? 

If "should", I would mention the circumstances that can prevent Symfony Flex from enabling the bundle, and if/how one can enable Solr manually.

Author: I didn't have the time to test and verify so that is why I as unsure. If anyone has install, remove solr, and try to require it to see what flex does. Or we omit the omit should, and take it as a bug if it turns out to be wrong 😉

Author: ```suggestion
    Symfony Flex will enable the bundle for you when installing the package.
```

!!! note "Solr versions"

>     For v3 `ezplatform-solr-search-engine` works with Solr 7 and 8, using an supported LTS versions of Solr is recommended.

Reviewer: ```suggestion
    v3 of `ezplatform-solr-search-engine` works with Solr 7 and Solr 8.
    It is recommended that you use a current major version of Solr.
```

Lucene does not refer to its versions as "LTS". They use terms like major, previous major, or EOL. Alternatively, we could say "It is recommended that you do not use an EOL version of Solr."

Author: The thing behind the recommendation:
1. 7.7 is given bug fixes still, while 8.x receives new features and sometimes possible breaks in minor releases
2. Platform.sh [supports 7.7, and 8.6](https://docs.platform.sh/configuration/services/solr.html), however they probably will take some time to add 8.7 and 8.8 when it arrives, so we try to avoid this kind of friction.

But maybe we should rather be specific to recommend versions:
- We test on
- available on p.sh

That would mean either 7.7 or 8.6 right now, and then we update it once 8.8 is out and available on psh.

Reviewer: "That would mean 7.7 or 8.6 right now and then we update it once 8.8 is out and available on psh." - makes perfect sense to me

Author: ```suggestion
    For v3 `ezplatform-solr-search-engine` works with Solr 7 and 8, using 7.7 or 8.6 of Solr is currently recommended.
```

##### Solr 7

> Download and extract Solr *(here using 7.7.2 as an example)*:

Reviewer: ```suggestion
Download and extract Solr, for example, in version 7.7.2:
```

> Download and extract Solr *(currently recommended with Solr 7.7LTS, so that is what example here is showing)*:

Reviewer: ```suggestion
Download and extract Solr *(a recommended major version is Solr 7.7.3)*:
```
7.7.3 is the last stable version of the 7.x series. 

To avoid confusion, I would use either this, or another tested and approved version, but give the exact version number, as assigned by Apache

Author: Major would refer to 7, so if we want to be specific patch release and say `7.7.3` I would maybe rather phrase it `*(7.7.3 + 8.6.3 tested and recommended at time of writing)*` But you end up having to update it quite often though, which is why I don't mention patch versions in doc usually but rather talk about which branches we support / recommend.

Reviewer: "at the time of writing" is assumed, we do not have to state that.

Can we say  *(the most recent release in 7.7 or 8.6 series is recommended)*

Author: ```suggestion
Download and extract Solr *(the most recent release in 7.7 or 8.6 series is recommended)*:
```
